File: P13 Roman to Integer/Solution.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dict = ["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]
        num = [1000,900,500,400,100,90,50,40,10,9,5,4,1]
        i, index, ret = 0, 0, 0
        while(i<len(s) and index<13):
            target = dict[index]
            cur = s[i:i+len(target)]
            if(cur==target):
                ret += num[index]
                i += len(target)
            else:
                index += 1
        return ret

File: P13 Roman to Integer/Solution.cpp
class Solution {
public:
    int romanToInt(string s) {
        string dict[] = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        int num[] = {1000,900,500,400,100,90,50,40,10,9,5,4,1};
        int i=0, index=0, ret=0;
        while(i<s.size() && index<13) {
            string target = dict[index];
            string cur = s.substr(i,target.size());
            if(cur==target) {
                ret += num[index];
                i += target.size();
            }
            else {
                index++;
            }
        }
        return ret;
    }
};